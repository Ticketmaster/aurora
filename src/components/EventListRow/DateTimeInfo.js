import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { colors, typography, spacing } from "../../theme";
import constants from "../../theme/constants";

import eventDateShape from "./shape";

const DateTimeInfoContainer = styled.div`
  width: 110px;
  padding-right: ${spacing.moderate};

  @media only screen and ${constants.device.mobile} {
    font-size: ${typography.size.kilo};
    width: 124px;
  }
`;

const DateTitle = styled.div`
  color: ${colors.heliotrope};
  font-size: ${typography.size.hecto};
  font-weight: ${typography.weight.semiBold};
  text-transform: uppercase;

  @media only screen and ${constants.device.mobile} {
    font-size: ${typography.size.kilo};
`;

const DateSubTitle = styled.div`
  color: ${colors.cascade.base};
  font-size: ${typography.size.hecto};
  white-space: nowrap;

  @media only screen and ${constants.device.mobile} {
    font-size: ${typography.size.kilo};
  }
`;

const DateTimeInfo = ({ date: { dateTitle = "", dateSubTitle = "" } = {} }) => (
  <DateTimeInfoContainer>
    <DateTitle>{dateTitle}</DateTitle>
    <DateSubTitle>{dateSubTitle}</DateSubTitle>
  </DateTimeInfoContainer>
);

DateTimeInfo.propTypes = {
  date: PropTypes.shape(eventDateShape).isRequired
};

export default DateTimeInfo;
