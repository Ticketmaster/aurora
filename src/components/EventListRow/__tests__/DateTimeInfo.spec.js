import React from "react";
import renderer from "react-test-renderer";

import DateTimeInfo from "../DateTimeInfo";

const mockDate = { dateTitle: "APR 23", dateSubTitle: "Thu, 8:00PM" };

describe("<DateTimeInfo />", () => {
  it("renders DateTimeInfo correctly", () => {
    const component = renderer.create(<DateTimeInfo date={mockDate} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
