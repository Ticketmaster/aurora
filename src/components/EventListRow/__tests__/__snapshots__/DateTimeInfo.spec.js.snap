// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateTimeInfo /> renders DateTimeInfo correctly 1`] = `
.c0 {
  width: 110px;
  padding-right: 9px;
}

.c1 {
  color: #c56bff;
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
}

.c2 {
  color: #69747c;
  font-size: 14px;
  white-space: nowrap;
}

@media only screen and (min-width:480px) {
  .c0 {
    font-size: 16px;
    width: 124px;
  }
}

@media only screen and (min-width:480px) {
  .c1 {
    font-size: 16px;
  }
}

@media only screen and (min-width:480px) {
  .c2 {
    font-size: 16px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    APR 23
  </div>
  <div
    className="c2"
  >
    Thu, 8:00PM
  </div>
</div>
`;
